sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -74,13 +74,13 @@ private object PostgresDialect extends JdbcDialect {
     case FloatType => Some(JdbcType("FLOAT4", Types.FLOAT))
     case DoubleType => Some(JdbcType("FLOAT8", Types.DOUBLE))
     case ShortType => Some(JdbcType("SMALLINT", Types.SMALLINT))
+    case ByteType => Some(JdbcType("SMALLINT", Types.SMALLINT))
```

Comment:
You could fold this case into the one above with `ShortType | ByteType` but no big deal